lib/core/src/Cardano/Pool/DB/Model.hs

```diff
@@ -430,16 +432,10 @@ mRollbackTo ti point = do
         | otherwise = Nothing
 
 mDelistPools :: [PoolId] -> ModelOp ()
-mDelistPools poolsToDelist =
-    modify #registrations
-        $ Map.mapWithKey
-        $ \(_, pid) a ->
-            if updateThis pid
-            then a {poolFlag = Delisted}
-            else a
-  where
-    updateThis p = p `Set.member` poolsToDelistSet
-    poolsToDelistSet = Set.fromList poolsToDelist
+mDelistPools = modify #delisted . Set.union . Set.fromList
```

Comment:
And we should also test this property. `delistPools` overwrites, it doesn't add.

Reply:
Aha, I didn't realize that `delistPools` was supposed to completely replace the set each time.

In that case, I think the `delistPools` name is a bit confusing, as it sounds as though it ought to augment the existing set, rather than replacing it.

Perhaps call the functions `putDelistedPools` and `readDelistedPools`?

Reply:
> That means we really want to overwrite here, not merge.

Fixed in b67c89fd8aed7b2349dd18ac4a0f8ed9e981823b.

Reply:
> And we should also test this property. delistPools overwrites, it doesn't add.

Fixed in b3f3d7ed45cde2a603b9fe26a6a916c89fd97132.